### PR TITLE
feat(PHC-4380): row selection to TableModule

### PR DESCRIPTION
**Pull request by shawnzhu**
It uses the [row selection API](https://tanstack.com/table/v8/docs/api/features/row-selection) from tanstack to simplify developers' learning curve.

This is an temporary solution before fully adopting the TanStack table API via #331

### CODE REVIEW

File: src/components/TableModule/TableModule.tsx
Comment (MMFane) [inline src/components/TableModule/TableModule.tsx:259]: Just want to make sure this <any> is the best course of action here

**Review by MMFane — COMMENTED** [review]
Looks good! A couple small pieces of feedback, and I'll approve after the few type errors blocking merge are resolved.

File: src/components/TableModule/types.ts
Comment (shawnzhu) [reply src/components/TableModule/types.ts:39]: the semantic of `Item` is one element out of array of data in the existing TableModule API, it was named _cell_ but functions as a _row_. Take below config for example:

https://github.com/lifeomic/chroma-react/blob/2c5973db869a67f1cede15ea2663f79032314dfb/src/components/TableModule/TableModule.stories.tsx#L70-L74

File: src/components/TableModule/TableModule.stories.tsx
Comment (shawnzhu) [reply src/components/TableModule/TableModule.stories.tsx:464]: I'd like to add something to the `TableModule.stories.tsx` but there's no `theme` so just need to figure out how to use this snippet

File: src/components/TableModule/TableModule.stories.tsx
Comment (shawnzhu) [reply src/components/TableModule/TableModule.stories.tsx:10]: I removed it since row selection works in the storybook w/o it. Since the selection box is totally userspace thing, I'd like to keep it simple by using a `<Checkbox/>` only

File: src/components/TableModule/TableModule.stories.tsx
Comment (MMFane) [reply src/components/TableModule/TableModule.stories.tsx:464]: I see some other components that leverage a theme in `/src/styles.` IMO, every table should have the same spacing here as a default and maybe it should be extensible, but I'm guessing most cases will want the basic spacing (or we may want to just enforce this)

File: src/components/TableModule/TableModule.tsx
Comment (shawnzhu) [reply src/components/TableModule/TableModule.tsx]: I remove this change but need such change in test code

File: src/components/TableModule/TableModule.stories.tsx
Comment (MMFane) [reply src/components/TableModule/TableModule.stories.tsx:463]: Since this is a temp solution before deprecating the table in favor of one that leverages tanStack's we can go without this for now

**Review by MMFane — APPROVED** [review]

**Pull request merged**
